### PR TITLE
Sparkjob - show and save UI ingress URL

### PR DESCRIPTION
**Pull request by urihoenig**

### CODE REVIEW

File: mlrun/runtimes/sparkjob.py [hunk 1]
```diff
@@ -232,7 +232,12 @@ def _run(self, runobj: RunObject, execution: MLClientCtx):
                             meta.name, driver, status
                         )
                     )
-                    logger.info("use .watch({}) to see logs".format(meta.name))
+                    resp = self.get_job(meta.name, meta.namespace)
+                    ui_ingress = get_in(resp, "status.driverInfo.webUIIngressAddress")
```
Comment (Hedingber) [inline mlrun/runtimes/sparkjob.py:236]: I don't really like this `get_in` thing, didn't do a refactor to remove it, but prefer not adding new usages of it, I prefer the classic way of doing it:
`resp.get('status', {}).get('driverInfo', {}).get('webUIIngressAddress')` (assuming all middle values are indeed support to be dicts)

File: mlrun/runtimes/sparkjob.py [hunk 2]
```diff
@@ -232,7 +232,12 @@ def _run(self, runobj: RunObject, execution: MLClientCtx):
                             meta.name, driver, status
                         )
                     )
-                    logger.info("use .watch({}) to see logs".format(meta.name))
```
Comment (Hedingber) [inline mlrun/runtimes/sparkjob.py:235]: Why you removed that ? it's not true ?

File: mlrun/runtimes/sparkjob.py [hunk 3]
```diff
@@ -232,7 +232,12 @@ def _run(self, runobj: RunObject, execution: MLClientCtx):
...
+                    resp = self.get_job(meta.name, meta.namespace)
+                    ui_ingress = get_in(resp, "status.driverInfo.webUIIngressAddress")
+                    if ui_ingress:
+                        runobj.status.status_text = "UI is available while the job is running: http://{}".format(
+                            ui_ingress
+                        )
```
Comment (Hedingber) [inline mlrun/runtimes/sparkjob.py:240]: Use python 3 string formatting
```suggestion
                        runobj.status.status_text = f"UI is available while the job is running: http://{ui_ingress}"
```
Please set the `runobj.status.host` as well so the address will be accessible from the Run object

**Review by Hedingber — CHANGES_REQUESTED** [review]

File: mlrun/runtimes/sparkjob.py (hunk 2 above)
Comment (urihoenig) [reply mlrun/runtimes/sparkjob.py:235]: it's both wrong and no longer relevant as the log is done on the mlrun-api server side

File: mlrun/runtimes/sparkjob.py (hunk 1 above)
Comment (urihoenig) [reply mlrun/runtimes/sparkjob.py:236]: not that I understand why, but done

File: mlrun/runtimes/sparkjob.py (hunk 3 above)
Comment (urihoenig) [reply mlrun/runtimes/sparkjob.py:240]: done

**Review by Hedingber — APPROVED** [review]

**Pull request merged**